From 31cf72681a6c9ad6cde2c8d7d8b7562443a691f2 Mon Sep 17 00:00:00 2001
From: Michal Simek <michal.simek@xilinx.com>
Date: Wed, 5 Dec 2012 16:50:09 +0100
Subject: [PATCH] arm: zynq: Remove XILINX_FIXED_DEVTREE_ADDR option

None is using this option. We are not using fixed
address anymore.

Signed-off-by: Michal Simek <michal.simek@xilinx.com>
Acked-by: John Linn <john.linn@xilinx.com>
---
 arch/arm/kernel/head.S     |    7 -------
 arch/arm/mach-zynq/Kconfig |    9 ---------
 2 files changed, 0 insertions(+), 16 deletions(-)

diff --git a/arch/arm/kernel/head.S b/arch/arm/kernel/head.S
index 3e13623..3db960e 100644
--- a/arch/arm/kernel/head.S
+++ b/arch/arm/kernel/head.S
@@ -91,13 +91,6 @@ ENTRY(stext)
  THUMB( it	eq )		@ force fixup-able long branch encoding
 	beq	__error_p			@ yes, error 'p'
 
-	/* 
-	 * if the device tree is expected at the fixed address
-	 * then load R2 to find the device tree at that address
-	 */
-#ifdef CONFIG_XILINX_FIXED_DEVTREE_ADDR
-	mov	r2,#0x1000000
-#endif
 #ifdef CONFIG_ARM_LPAE
 	mrc	p15, 0, r3, c0, c1, 4		@ read ID_MMFR0
 	and	r3, r3, #0xf			@ extract VMSA support
diff --git a/arch/arm/mach-zynq/Kconfig b/arch/arm/mach-zynq/Kconfig
index 5c3b91a..4c241f0 100644
--- a/arch/arm/mach-zynq/Kconfig
+++ b/arch/arm/mach-zynq/Kconfig
@@ -17,15 +17,6 @@ config ZYNQ_EARLY_UART_EP107
 	  the UART that is based on a 33.333 MHz clock divided down by 63.  Note that
 	  this only affects early printk.
 
-config XILINX_FIXED_DEVTREE_ADDR
-	bool "Device Tree At Fixed Address"
-	default n
-	depends on OF
-	help
-	  Select if you want the kernel to assume a device tree is located at a
-	  fixed address of 0x1000000 on kernel boot.  This allows the go command
-	  in u-boot to be used since the bootm command is slow in the EP107.
-
 config XILINX_L1_PREFETCH
 	bool "L1 Cache Prefetch"
 	default y
-- 
1.7.1

